test/ping-pong/proposer/src/app.mjs

```diff
@@ -0,0 +1,32 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import fileUpload from 'express-fileupload';
+import * as proposer from './routes/proposer.mjs';
+
+const app = express();
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
+  next();
+});
+
+app.use(cors());
+app.use(bodyParser.json({ limit: '2mb' }));
+app.use(bodyParser.urlencoded({ limit: '2mb', extended: true }));
+app.use(
+  fileUpload({
+    createParentPath: true,
+  }),
+);
+
+app.get('/healthcheck', (req, res) => {
+  res.sendStatus(200);
+});
+
+app.use('/proposer', proposer.router);
+
+function setOptimistUrl(url) {
```

Comment:
proposer already is setting this URL when it calls 

```
new Nf3(ethereumSigningKey, {
    web3WsUrl,
    clientApiUrl,
    optimistApiUrl,
    optimistWsUrl,
  });
```

Reply:
My issue is that i need to access the optimist URL from a file within the proposer structure (proposer/routes/proposer.mjs) so that the proposer forwards the call to optimist. This route file doesn't have access to the URL, and that is why i am passing it down. 
If i don't do it like this, I fail to understand how to do it. Do you have a suggestion?

Reply:
My suggestion is to implement `sendOffchainTransaction` in SDK similar to the other proposer functions that SDK implements already `registerProposer`, `deregisterProposer`, `changeCurrentProposer`, `getCurrentProposer`, `getProposers`. 

The proposer's route file will then call this function from SDK. 
The objective of the SDK is to abstract the logic of implementation from a proposer.
The proposer will only have to think about calling the relevant SDK with configuration that they prefer by having configureable variables. One example I use to explain it easily is how a user uses geth JSON RPC to configure their block chain node without implementing any functional logic for the geth node. In our case, it will be SDK based REST API. If the SDK is accessed from route file or proposer.mjs file, it is still the proposer that is using it.